authorize/evaluator/headers_evaluator.go

```diff
@@ -54,6 +57,45 @@ type HeadersResponse struct {
 	Headers http.Header
 }
 
+var variableSubstitutionFunctionRegoOption = rego.Function2(&rego.Function{
+	Name: "pomerium.variable_substitution",
+	Decl: types.NewFunction(
+		types.Args(
+			types.Named("input_string", types.S),
+			types.Named("replacements",
+				types.NewObject(nil, types.NewDynamicProperty(types.S, types.S))),
+		),
+		types.Named("output", types.S),
+	),
+}, func(bctx rego.BuiltinContext, op1 *ast.Term, op2 *ast.Term) (*ast.Term, error) {
+	inputString, ok := op1.Value.(ast.String)
```

Comment:
I've tried to follow the example of [getGoogleCloudServerlessHeadersRegoOption](https://github.com/pomerium/pomerium/blob/5568606f03023c1f936ed70b5a41fe3679ba6426/authorize/evaluator/google_cloud_serverless.go#L37C1-L59) here, but I don't completely understand whether all these type checks are needed (given the `Decl` type above).